### DISCUSSION TITLE
FragmentType mock for story book and tests

### DISCUSSION THREAD

**Discussion by matheusrocha89** [dotansimha/graphql-code-generator#8624] [Q&A] [answered]
I am using the `client-preset` with `data-masking`. 
But I am struggling now to mock the data passed to the props of the component that are `FragmentType` on storybook or tests. Even if I create an object with the properties it says it's not compatible. The only solution I found was casting the object `as any` when I am passing it. 

What do you people do in these situations? 

<img width="696" alt="image" src="https://user-images.githubusercontent.com/3718366/203041337-7ccb250a-02ba-468d-b1c3-1454b95daeb6.png">

**Comment by charlypoly** [comment]
Hi @matheusrocha89,

It would be best if you did not use the `FragmentType<T>` type helper here but the generated fragment type directly, as explained here: https://github.com/dotansimha/graphql-code-generator/discussions/8554#discussioncomment-4136414.

**Reply by matheusrocha89** [reply]
But in all examples it uses the `FragmentType` 
https://the-guild.dev/blog/unleash-the-power-of-fragments-with-graphql-codegen#compose-fragments-for-ui-components
https://the-guild.dev/graphql/codegen/docs/guides/react-vue#writing-graphql-fragments

So is this wrong?

**Reply by matheusrocha89** [reply]
I can't do that, because when I do the `useFragment` have type issues now.

**Reply by matheusrocha89** [reply]
So or I use the FragmentType and useFragment or I use the generated Type of my fragment and do not use data masking with useFragment?

**Reply by charlypoly** [reply]
Could you provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) using our [GitHub](https://github.com/dotansimha/graphql-code-generator-issue-sandbox-template), [Stackblitz](https://stackblitz.com/github/dotansimha/graphql-code-generator-issue-sandbox-template) or [CodeSandbox](https://codesandbox.io/s/github/dotansimha/graphql-code-generator-issue-sandbox-template) template?

That would help me in providing you a full example 🙏🏼

**Reply by matheusrocha89** [reply]
Sure, check this example, I just created a test to show you, but it's the same for storybook, test or anything related that I need to mock the data for that.

https://github.com/matheusrocha89/codege-fragment-example/blob/main/src/components/ProfileName/ProfileName.test.tsx

**Reply by charlypoly** [reply]
@matheusrocha89, I'm working on a helper to help with that: https://github.com/dotansimha/graphql-code-generator/pull/8657

**Comment by charlypoly** [comment, marked as answer]
Hi @matheusrocha89,

You can now refer to the newly created Fragment Masking documentation: https://the-guild.dev/graphql/codegen/plugins/presets/preset-client#fragment-masking-and-testing

**Reply by matheusrocha89** [reply]
Nice, thanks @charlypoly , already updated the example project.
